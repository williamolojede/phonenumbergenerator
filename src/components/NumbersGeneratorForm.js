import React, { Component } from 'react'

export default class NumbersGeneratorForm extends Component {
  inputEl = React.createRef();
  
  handlesubmit = (e) => {
    const total = Number(this.inputEl.current.value)
    e.preventDefault();
    const numbers = this.generateRandomNumbers(total);
    this.props.handleGeneratedNumbers(numbers);
  }

  generateRandomNumber = () => {
    return Math.floor(Math.random() * Math.pow(10, 8)) + Math.pow(10, 8)
  }

  generateRandomNumbers = (count) => {
    const numbers = [];
    for (let i = 0; i < count; i += 1) {
      numbers.push(this.generateRandomNumber());
    }
    return numbers;
  }

  render() {
    return (
      <form onSubmit={this.handlesubmit} >
        <p className="uppercase mb-6 text-center text-grey-dark">Enter the number of phone numbers to be generated</p>

        <div className="flex justify-between">
          <input 
            type="number" 
            ref={this.inputEl}
            className="border-b-2 border-grey-darker text-5xl w-32 text-center"
            required
            max="500"
            min="10"
          />


          <button type="submit" className="border-2 border-grey-darker p-4 w-1/2 text-xl font-bold uppercase text-grey-darker">Generate</button>
        </div>
      </form>
    )
  }
}