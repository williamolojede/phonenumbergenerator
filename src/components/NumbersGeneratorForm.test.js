import React from 'react'
import { mount } from 'enzyme';

import NumbersGeneratorForm from './NumbersGeneratorForm'

const setup = () => {
  const props = {
    handleGeneratedNumbers:  jest.fn(),
  }
  const wrapper = mount(<NumbersGeneratorForm {...props}/>);

  return {
    props,
    wrapper
  }
}

it('renders without crashing', () => {
  const { wrapper } = setup();

  expect(wrapper).toMatchSnapshot();
});

it('generates a 9 digits random number', () => {
  const { wrapper } = setup();
  const number =wrapper.instance().generateRandomNumber();
  expect(String(number).length).toBe(9);
})

it('generates a four 9 digits random number', () => {
  const { wrapper } = setup();
  const numbers = wrapper.instance().generateRandomNumbers(4);
  expect(numbers.length).toBe(4);
  numbers.forEach(num => expect(String(num).length).toBe(9))
});

it('generates random numbers when form is submitted', () => {
  const { wrapper, props } = setup();
  const Form = wrapper.find('form');
  const preventDefault = jest.fn()

  Form.simulate('submit', {
    preventDefault
  })
  expect(preventDefault).toHaveBeenCalled();
  expect(props.handleGeneratedNumbers).toHaveBeenCalled();
})