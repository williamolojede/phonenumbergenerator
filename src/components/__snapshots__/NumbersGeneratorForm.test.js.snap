// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<NumbersGeneratorForm
  handleGeneratedNumbers={[MockFunction]}
>
  <form
    onSubmit={[Function]}
  >
    <p
      className="uppercase mb-6 text-center text-grey-dark"
    >
      Enter the number of phone numbers to be generated
    </p>
    <div
      className="flex justify-between"
    >
      <input
        className="border-b-2 border-grey-darker text-5xl w-32 text-center"
        max="500"
        min="10"
        required={true}
        type="number"
      />
      <button
        className="border-2 border-grey-darker p-4 w-1/2 text-xl font-bold uppercase text-grey-darker"
        type="submit"
      >
        Generate
      </button>
    </div>
  </form>
</NumbersGeneratorForm>
`;
